src/CredentialsWrapper.php

```diff
@@ -232,6 +254,26 @@ public function getAuthorizationHeaderCallback($audience = null)
         };
     }
 
+    /**
+     * Verify that the expected universe domain matches the universe domain from the credentials.
+     */
+    private function checkUniverseDomain()
```

Comment:
We don't seem to have a way to differentiate UD passed from credentials file or was user supplied to check for the following scenario:

If a user passes in the universeDomain client option WITH custom credentials which do not implement GetUniverseDomainInterface, the check would not occur, increasing the chance of credentials not honoring UD config.

Do you think we should add this in comments?

Reply:
Sorry for the confusion. This sentence was in consideration of a _different_ implementation that was rejected.

```
If a user passes in the universeDomain client option WITH custom credentials which do not implement GetUniverseDomainInterface, the check would not occur, increasing the chance of credentials not honoring UD config.
```

I will add a test for the scenario above, to ensure that it's not the case.

Reply:
Looks like [I already added a test for this!](https://github.com/googleapis/gax-php/pull/502/files#diff-d7d06c054fa4a4bc3263a0062e0a341cac26d42d6958662fb79f699a973de523R262)